labeling/labelers.go

```diff
@@ -0,0 +1,42 @@
+package labeling
+
+import (
+	"strings"
+
+	bsky "github.com/bluesky-social/indigo/api/bsky"
+)
+
+// simple record labeling (without pre-fetched blobs)
+type SimplePostLabeler interface {
+	labelPost(p bsky.FeedPost) []string
```

Comment:
Can you emit a label for an actor based on one of their posts?

Reply:
Yes. This initial version only outputs labels on the specific record, but the overall architecture allows outputting labels on either on records or on repos (DIDs). For example, if this was the Nth label on a record in the same repo we could have logic that would output a repo-level label.

An open question is how/whether to label pieces of infrastructure, like a PDS. There is an idea of a way to label DNS names (so, a PDS at a stable domain). If infrastructure has stable DIDs we could label those, but I don't think we've thought that through all the way (aka, how to we prevent infra from just cycling through DIDs; do we really want to require stable DIDs for infra in the first place).

Reply:
Okay, separate question is how can i load the blobs from this post in order to use them as context for label decisions?
Also should probably wire contexts through all of this (esp if we're going to be making external API calls at some point)

Reply:
Fetching blobs is presumably going to be a multi-step process of resolving the DID to PDS (cachable), and then fetching the blob with `getBlob`. The atproto branch with getBlob implementation hasn't landed, it is here: https://github.com/bluesky-social/atproto/pull/606

I did a manual rebase locally of that branch to test things, and ran in to the mentioned issue with records seemingly missing blob metadata.

You can see my branch with external blob (image) fetches here: https://github.com/bluesky-social/indigo/pull/61